src/blueapi/worker/task_worker.py

```diff
@@ -181,8 +182,8 @@ def mark_task_as_started(event: WorkerEvent, _: str | None) -> None:
                 task_started.set()
 
         LOGGER.info(f"Submitting: {trackable_task}")
+        sub = self.worker_events.subscribe(mark_task_as_started)
```

Comment:
Should: Moving this out of the `try` block means a network error could potentially bring down the worker subprocess

Reply:
from the first look on it, to add a callback on a potential event is not a network event. 

it is only the following line:

```python
            self._task_channel.put_nowait(trackable_task)
            
```

that does something that looks like network logic

Reply:
That's not network logic, that's adding to an internal queue. However, it is not just network logic you need to worry about: Anything that can throw an exception should go inside the `try` block, since we don't want the subprocess to crash and leave the main process spinning.

Reply:
> from the first look on it, to add a callback on a potential event is not a network event.


as I expected. if the first look was wrong, it means that the variable and file naming is not descriptive enough...

Reply:
why adding to an internal queue would fail?

Reply:
`put_nowait` fails if the queue is full. `put` does not fail but blocks until the queue is no longer full.

https://docs.python.org/3/library/asyncio-queue.html

Reply:
why would the queue be full?

Reply:
[It has a max capacity of 1 item at a time](https://github.com/DiamondLightSource/blueapi/blob/c9c2d449463cc0810452cf24b53f267ed437f1ba/src/blueapi/worker/task_worker.py#L103)

Reply:
that's a material reason for it, what is the teleological reason?

Reply:
The queue is actually a channel between 2 threads, most programming languages have a separate abstraction for this but Python reuses `Queue`. So it's not being used as a queue really, but more as a box in which to put the "current" task and nothing else. So the calling thread can only ever submit one plan at a time.

Reply:
I added a comment explaining that part.

Still I think the discussion got sidetracked a bit as
>         sub = self.worker_events.subscribe(mark_task_as_started)
is not the `put_nowait` line

Reply:
You're right, I don't think that line could cause a network error, but it could still raise an exception, so why was it moved out of the `try` block?

Reply:
because it's just adding a callback

Reply:
So if anything in `_cycle` raises an uncaught error it will kill the worker thread and blueapi will need a manual restart, which is why I prefer to play it safe and just catch and log _any_ exception in the cycle loop. That code won't currently raise an exception, but it could be modified in the future such that it does, and the person modifying it won't necessarily know about this line. Please put it back.

Reply:
ok, I'll include information from your comment for clarity